internal/dnstransport/dnsoverhttps/dnsoverhttps_test.go

```diff
@@ -1,22 +1,18 @@
-package dnsoverhttps_test
+package dnsoverhttps
```

Comment:
The API exposed too much to let tests be implemented. I've made all these unnecessarily public fields private and changed the tests to be inside rather than outside the package (which is the line you commented on)